engines/suppliers/app/controllers/suppliers/api/v1/providers_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Suppliers
+  module Api
+    module V1
+      class ProvidersController < ApplicationController
+        before_action :authenticate
+        before_action :load_group, only: [:index]
+        before_action :load_producer, only: [:show]
+
+        # GET /api/v1/providers?group_id=
+        #
+        # You must pass `group_id` param as filter
+        #
+        def index
+          producer_ids = ::Suppliers::Supplier.where(group_id: @group.id).pluck(:producer_id)
+          providers = ::Suppliers::Producer.where(id: producer_ids)
+
+          render json: ::Suppliers::ProvidersSerializer.new(providers)
+        end
+
+        # GET /api/v1/providers/:id
+        #
+        def show
+          render json: ::Suppliers::ProviderSerializer.new(@producer)
+        end
+
+        private
+
+        def load_group
+          @group = ::Suppliers::Group.find_by_id(params[:group_id])
+
+          return head :not_found unless @group
+
+          authorize @group
+        end
+
+        def load_producer
+          @producer = ::Suppliers::Producer.find_by_id(params[:id])
+
+          return head :not_found unless @producer
+
+          authorize @producer
```

Comment:
In this specific case the `show` and `index` endpoints in `providers` controller use the `suppliers` relationship. The `producers` controller `show` and `index` use the `memberships` relation instead.

The first relation, `suppliers`, tells you if the user can use that `producer` as a provider and place orders.
The second one, `memberships`, tells you if the user ha admin authority on the given `producer`. Useful for instance to change the name of the producer or add a product associated to it.

The data model is a bit complex here since `producer` model has two kind of "join tables". And we need that since one (`producers_memberhips`) represents the ownership of the `producer` (with `users` or `groups`), while the other (`suppliers`) represents the relationships as a provider between a `producer` ad a `group`.